packages/gatsby-dev-cli/src/local-npm-registry/check-deps-changes.js

```diff
@@ -0,0 +1,89 @@
+const fs = require(`fs-extra`)
+const _ = require(`lodash`)
+const { getMonorepoPackageJsonPath } = require(`./utils`)
+
+function difference(object, base) {
+  function changes(object, base) {
+    return _.transform(object, function(result, value, key) {
+      if (!_.isEqual(value, base[key])) {
+        result[key] =
+          _.isObject(value) && _.isObject(base[key])
+            ? changes(value, base[key])
+            : value
+      }
+    })
+  }
+  return changes(object, base)
+}
+
+const checkDepsChanges = ({ newPath, packageName, root, isInitialScan }) => {
+  const monoRepoPackageJsonPath = getMonorepoPackageJsonPath({
+    packageName,
+    root,
+  })
+  const monorepoPKGjson = JSON.parse(fs.readFileSync(monoRepoPackageJsonPath))
+
+  let localPKGjson
+  try {
+    localPKGjson = JSON.parse(fs.readFileSync(newPath))
+  } catch {
+    // caught if there is no local package - so we still need to install deps
+    // this is nice because devs won't need to do initial package installation - we can handle this.
+    console.log(
+      `'${packageName}' doesn't seem to be installed. Will install it`
+    )
+    return true
+  }
+
+  const areDepsEqual = _.isEqual(
+    monorepoPKGjson.dependencies,
+    localPKGjson.dependencies
+  )
+
+  if (!areDepsEqual) {
+    const diff = difference(
+      monorepoPKGjson.dependencies,
+      localPKGjson.dependencies
+    )
+
+    const diff2 = difference(
+      localPKGjson.dependencies,
+      monorepoPKGjson.dependencies
+    )
+
+    let needPublishing = false
+    const depChangeLog = _.uniq(Object.keys({ ...diff, ...diff2 }))
+      .reduce((acc, key) => {
+        if (
```

Comment:
need to add handling of temporary file changes when I set dependency to `gatsby-dev` dist-tag - as this cause random and confusing console output that you can't publish/install things